src/format/parse.rs

```diff
@@ -1040,3 +1298,135 @@ fn test_issue_1010() {
+    assert_eq!(parse("2023-06-9 "), Err(INVALID));
+    assert_eq!(parse("23-06-09 "), Err(INVALID));
+
+    // ordinal date, basic format
+    assert_eq!(parse("2023160 "), Ok((NaiveDate::from_yo_opt(2023, 160).unwrap(), " ")));
+    // ordinal date, extended format
+    assert_eq!(parse("2023-160 "), Ok((NaiveDate::from_yo_opt(2023, 160).unwrap(), " ")));
+    // ordinal date, basic format, expanded representation
+    assert_eq!(parse("-2023160 "), Ok((NaiveDate::from_yo_opt(-2023, 160).unwrap(), " ")));
+    assert_eq!(parse("+2023160 "), Ok((NaiveDate::from_yo_opt(2023, 160).unwrap(), " ")));
+    // ordinal date, extended format, expanded representation
+    assert_eq!(parse("-2023-160 "), Ok((NaiveDate::from_yo_opt(-2023, 160).unwrap(), " ")));
+    assert_eq!(parse("+2023-160 "), Ok((NaiveDate::from_yo_opt(2023, 160).unwrap(), " ")));
+    assert_eq!(parse("+02023-160 "), Ok((NaiveDate::from_yo_opt(2023, 160).unwrap(), " ")));
+    assert_eq!(parse("+12023-160 "), Ok((NaiveDate::from_yo_opt(12023, 160).unwrap(), " ")));
+    // ambiguous, interpreted as calendar date
+    assert!(parse("+02023160 ").is_err());
+    assert!(parse("+12023160 ").is_err());
+    // No padding
+    assert_eq!(parse("2023-16 "), Err(INVALID));
+    assert_eq!(parse("2023-1 "), Err(INVALID));
+    assert_eq!(parse("23-160 "), Err(INVALID));
+
+    let from_isoywd_opt = NaiveDate::from_isoywd_opt;
+    // week date, basic format
+    assert_eq!(parse("2023W235 "), Ok((from_isoywd_opt(2023, 23, Fri).unwrap(), " ")));
+    // week date, extended format
+    assert_eq!(parse("2023-W23-5 "), Ok((from_isoywd_opt(2023, 23, Fri).unwrap(), " ")));
+    // week date, basic format, expanded representation
+    assert_eq!(parse("-2023W235 "), Ok((from_isoywd_opt(-2023, 23, Fri).unwrap(), " ")));
+    assert_eq!(parse("+2023W235 "), Ok((from_isoywd_opt(2023, 23, Fri).unwrap(), " ")));
+    assert_eq!(parse("+02023W235 "), Ok((from_isoywd_opt(2023, 23, Fri).unwrap(), " ")));
+    assert_eq!(parse("+12023W235 "), Ok((from_isoywd_opt(12023, 23, Fri).unwrap(), " ")));
+    // calendar date, extended format, expanded representation
+    assert_eq!(parse("-2023-W23-5 "), Ok((from_isoywd_opt(-2023, 23, Fri).unwrap(), " ")));
+    assert_eq!(parse("+2023-W23-5 "), Ok((from_isoywd_opt(2023, 23, Fri).unwrap(), " ")));
+    assert_eq!(parse("+02023-W23-5 "), Ok((from_isoywd_opt(2023, 23, Fri).unwrap(), " ")));
+    assert_eq!(parse("+12023-W23-5 "), Ok((from_isoywd_opt(12023, 23, Fri).unwrap(), " ")));
+    // mixed basic and extended format
+    assert_eq!(parse("2023-W235 "), Err(INVALID));
+    assert_eq!(parse("202306-W235 "), Err(INVALID));
+    assert_eq!(parse("-2023-W235 "), Err(INVALID));
+    // No padding
+    assert_eq!(parse("2023-W25 "), Err(INVALID));
+    assert_eq!(parse("23-W23-5 "), Err(INVALID));
+    // Year is interpreted as `iso_year`
+    assert_eq!(parse("2022-W52-7 "), Ok((NaiveDate::from_ymd_opt(2023, 1, 1).unwrap(), " ")));
+}
+
+#[cfg(test)]
```

Comment:
I'm pretty sure that if it's marked `#[test]` then the `#[cfg(test)]` is redundant.

> The `test` attribute marks a function to be executed as a test. These functions are only compiled when in test mode.

https://doc.rust-lang.org/reference/attributes/testing.html#the-test-attribute

Reply:
It was temporary to silence the CI until I could rebase on https://github.com/chronotope/chrono/pull/1142 :smile: